packages/kbn-elastic-agent-utils/src/agent_guards.ts

```diff
@@ -7,32 +7,50 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
-import { JAVA_AGENT_NAMES, OPEN_TELEMETRY_AGENT_NAMES, RUM_AGENT_NAMES } from './agent_names';
+import {
+  ANDROID_AGENT_NAMES,
+  IOS_AGENT_NAMES,
+  JAVA_AGENT_NAMES,
+  OPEN_TELEMETRY_AGENT_NAMES,
+  RUM_AGENT_NAMES,
+} from './agent_names';
 
 import type {
+  AndroidAgentName,
+  IOSAgentName,
   JavaAgentName,
   OpenTelemetryAgentName,
   RumAgentName,
   ServerlessType,
 } from './agent_names';
 
+export function hasOpenTelemetryPrefix(agentName?: string, language: string = '') {
+  if (!agentName) {
+    return false;
+  }
+
+  return (
+    agentName.startsWith(`opentelemetry/${language}`) || agentName.startsWith(`otlp/${language}`)
```

Comment:
if `agentName` doesn't end with `/` and `language` is `''`, this logic will fail. Perhaps we could add a condition to cover that?

Reply:
As this is only for OpenTelemetry agents, we will always have the `/` ending. Regarding language, is to be able to pass it as parameters and avoid duplicating code

Reply:
> As this is only for OpenTelemetry agents, we will always have the / ending

I'm not sure about that, actually! I think there are still cases that we have just `opentelemetry` or just `otlp` as the `agent.name`.

Reply:
Nevermind, saw the comment below!